src/lib/badge/_badge-theme.scss

```diff
@@ -0,0 +1,177 @@
+@import '../core/theming/palette';
+@import '../core/theming/theming';
+@import '../core/typography/typography-utils';
+
+$mat-badge-font-size: 12px;
+$mat-badge-font-weight: 600;
+$mat-badge-size: 22px !default;
+
+.mat-badge {
+  position: relative;
+}
+
+.mat-badge:after,
+.mat-badge-icon,
+.mat-badge-text,
+.mat-badge-svg-icon {
+  position: absolute;
+  text-align: center;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  justify-content: center;
+  align-content: center;
+  align-items: center;
+}
+
+.mat-badge:after {
+  width: $mat-badge-size;
+  height: $mat-badge-size;
+  border-radius: 50%;
+  content: '';
+  z-index: 1;
+}
+
+.mat-badge-text,
+.mat-badge .mat-badge-icon {
+  z-index: 2;
+  font-size: 16px;
```

Comment:
This should be in the typography mixin and the value, ideally, should come from the typography breakpoints.

Reply:
Sure, I've moved it inside that mixin.

This size is independent from the font-size of the page since its specifically designed for this fixed width/height of the badge.